.github/workflows/build.yml

```diff
@@ -1,14 +1,16 @@
-name: Build Docker image
-
-on: [push]
+name: Build
 
+on:
+  push:
+    paths-ignore:
+      # don't run this if docs are the only thing affected
+      - "docs/**"
 env:
   PARABOL_DOCKERFILE: ./docker/parabol-ubi/docker-build/dockerfiles/pipeline.dockerfile
   PARABOL_BUILD_ENV_PATH: docker/parabol-ubi/docker-build/environments/pipeline
-  DEPENDENCIES_CACHE: cache-node-modules
 
 jobs:
-  build-and-push:
+  build:
     runs-on: ubuntu-4core
```

Comment:
This script is serving 2 purposes: 
- making sure a commit is good to merge upstream
- building the docker image

By forking the logic like this, we can reduce the wait time to merge by a couple minutes, which is huuuge for devs!

Reply:
I'm fine with that, I was just trying to share that either way we were using the paid runners as everything `runs-on: ubunutu-4core`.

Reply:
One more note: I don't think the title of the step is clear enough `Build cheap` and `Build prod`. We will build using `--no-deps` in non-prod scenarios, multiple times.

As the we are building in one way or the other depending on `DOCKERIZE` I would change those titles to `Build for Docker` and `Build for non Docker`. Which isn't even true, because we could build the app using `--no-deps` for non Docker uses cases, but well, at least is clearer.

WDYT?

Reply:
Could you tell me more why we'd ever use `--no-deps` if we're not going to deploy?

As far as I know, there are 2 reasons to run this script:

1. Make sure the new code I committed is well written & doesn't break existing features. This is useful for devs so we can merge code with confidence. Speed here is very important because I shouldn't merge my PR until this script completes.
2. Same as above, PLUS build a docker image. This is useful for deploying a new version. Minimizing disk size is more important than speed here.

If there's another use case, let me know!

Reply:
I didn't have any particular scenario in mind. I still don't have any. Just saying that `build cheap` isn't clear for me. I think that if I would be a new person checking it or even myself months from now, I wouldn't get why it is called `cheap`.

That said, this is clearly a minor thing that has no impact on how the pipeline works! We can just skip this for now!

Reply:
Makes total sense! Renamed to:
- Build for testing
- Build for deploying

_technically_ i guess there's a difference between dockerizing & deploying, but here i'm treating them both the same.

Reply:
I like that way more. Thanks Matt!